system-addon/test/functional/mochitest/browser_topsites_section.js

```diff
@@ -20,6 +20,8 @@ test_newtab({
   async before({pushPrefs}) {
     // The pref for TopSites is empty by default.
     await pushPrefs(["browser.newtabpage.activity-stream.default.sites", "https://www.youtube.com/,https://www.facebook.com/,https://www.amazon.com/,https://www.reddit.com/,https://www.wikipedia.org/,https://twitter.com/"]);
+    await pushPrefs(["browser.newtabpage.activity-stream.enabled", false]);
```

Comment:
If we're going for a hacky way to force a refresh, we probably shouldn't do it by completely restarting activity stream:
1) this particular pref might go away soon as activity stream will just always be enabled
2) the full shutdown then startup of activity stream could lead to other issues similar to the issues related to "restarting" places data

We should comment that we're trying to force top sites to refresh if there's a better way to do this down the line.

A potentially more limited action that hopefully is safe is PLACES_LINK_BLOCKED. MIGRATION_COMPLETED could be an alternative, but the only way to trigger that would be to `notifyObservers` of the appropriate event, but other observers might do unexpected stuff…

Reply:
I was expecting TopSitesFeed to behave more like the Sections. Shouldn't it react to `showTopSites` pref changes?  
https://github.com/mozilla/activity-stream/blob/8e5857c1f8b102c410892708dba9bb6f6bc3fcf6/system-addon/lib/SectionsManager.jsm#L330-L335

Reply:
I suppose toggling the feed should be better than toggling all of activity stream. If it passes tests without intermittents, I guess it's okay with comments.